brewtils/errors.py

```diff
@@ -112,3 +115,74 @@ class BGConflictError(BrewmasterRestError):
 class BGNotFoundError(BrewmasterRestError):
     """Error Indicating a 404 was raised on the server"""
     pass
+
+
+def parse_exception_as_json(exc):
+    """
+    Attempt to parse an Exception to a JSON string.
+
+    If the exception has a single argument, no attributes, and the attribute
+    can be converted to a valid JSON string, then that will be returned.
+
+    Otherwise, a string version of the following form will be returned:
+
+    {
+        "message": "",
+        "arguments": [],
+        "attributes": {}
+    }
+
+    Where "message" is just str(exc), "arguments" is a list of all the
+    arguments passed to the exception attempted to be converted to a valid
+    JSON string, and "attributes" are the attributes of the exception class.
+
+    If parsing fails at all, then a simple str() will be applied either the
+    argument or attribute value.
+
+    Note:
+        On python version 2, errors with custom attributes do not list those
+        attributes as arguments.
+
+    Args:
+        exc (Exception):
+
+    Returns:
+        A valid JSON string representing (the best we can) the exception.
+
+    """
+    if not isinstance(exc, Exception):
+        raise ValueError("Attempted to parse a non-exception as JSON.")
```

Comment:
Should we be raising an exception in this method?

Reply:
I think so.